File: Roadmap/02 - FUNCIONES Y ALCANCE/javascript/raulG91.js

//Simple
function sayHello(){

    console.log("Hello World!");
}
console.log("Function without parameter and without return");
sayHello();
//With retrun a not parameter
function sayHello2(){
    return "Hello Team!";
}
console.log("Function without parameter and with return");
console.log(sayHello2());
//With parameters not return
function sumNumber(a,b){
    let result = a+b;
    console.log(result)
}
console.log("Function with parameters and not return");
sumNumber(2,3);

//With parameters and return
function multiplyNumbers(a,b){
    return a*b;
}
console.log("Function with parameters and return");
let result = multiplyNumbers(2,3);
console.log(result);

//Function with default parameters
function default_parameter(phrase="Hello world"){

    console.log(phrase);
}

console.log("Function with default parameter");
default_parameter();
default_parameter("This is a function");

//Nested functions
function main_function(){

    function nested_function(){

        console.log("Hello from nested function");
    }

    nested_function();
}

console.log("Nested function");
main_function();

// Javascript functions
let a = 3;
console.log("Function typeof");
console.log(typeof(a));

let string = "This is an string";
let array = string.split(" ");
console.log("Split function");
console.log(array);

let variable = 6

function modify_variable(){

    let variable = 10;
    console.log(variable);
}

modify_variable();
console.log("Original variable " + variable);

//Extra exercise

function print_numbers(string1,string2){

    let numbers = 0;

    for(let i=1;i<=100;i++){

        if(i%3 == 0 && i %5 == 0){
            console.log(string1+string2);
        }
        else if (i%3==0){
            console.log(string1);
        }
        else if (i%5==0){
            console.log(string2);
        }
        else{
            console.log(i);
            numbers++;
        }
    }

    return numbers;
}

let numbers = print_numbers("fizz","buzz");

console.log(numbers);

File: Roadmap/02 - FUNCIONES Y ALCANCE/python/raulG91.py
#Function without parameters and without return
def say_hello():
    print("Function without parameters and return!")
    
say_hello()    

#Function with parameters and return 
def sum_numbers(a,b):
    return a + b

print("Result function with parameters and return: " + str(sum_numbers(3,5)))

#Function with parameters but no return

def print_sum_numbers(a,b):
    print("Result is: " + str(a+b))
    
print("Function with parameters but no return :")
print_sum_numbers(2,3)    

#Function with default parameter

def default_parameter(phrase = "Default parameter"):
    return phrase
    
print("Function with default parameter: ",default_parameter())
print("Function with default parameter: ",default_parameter("Modify default parameter"))


#Lambda function

f = lambda: "Message from a lambda function"

print("Lambda function : " + str(f()) )   

#Function with multiple arguments

def multiple_arguments(*args):
    for pos,argument in enumerate(args):
        print("Argument in pos: " + str(pos) +" "+ str(argument))
print("Function with multiple arguments: ")
multiple_arguments("Hello")
multiple_arguments("Open","Close")        
        
def multiple_arguments_keyword(**kargs):
    for key in kargs:
        print("Parameter key: " + str(key) + " parameter: " + str(kargs[key]))
        
multiple_arguments_keyword(string_parameter="Hi", number_parameter=5)       


def wrap_function():
    
    a = 5 

    def nested_function():
        return a**2
    
    result = nested_function()
    return result

print("Result of wrap/nested function: " + str(wrap_function()))

GLOBAL_VARIABLE = 3

number_variable = 3

def modify_variable():
    number_variable = 6
    print("Value is: ",number_variable)

modify_variable()
print("Value is: ", number_variable)    

#Language functions

text = "This is a text"
print("Funcrion Len:" + str(len(text)))

print("Uppercase function: ",  text.upper())



number_list = [5,2,6,16,7,3]
filtered_list = filter(lambda x: x%2 == 0,number_list)
print("Filter and list function", list(filtered_list))

def extra_exercise(arg1,arg2):
    
    counter = 0
    
    for number in range(1,101):
        
        if number % 3 == 0 and number % 5 == 0:
            print(arg1+arg2)
        elif number % 3 == 0:
            print(arg1) 
        elif number % 5 == 0:
            print(arg2)
        else:
            print(number)
            counter+=1
    return counter

numbers = extra_exercise("fizz","buzz")
print(numbers)        
            
    